tsk/fs/hfs.c

```diff
@@ -1469,9 +1469,10 @@ hfs_follow_hard_link(HFS_INFO * hfs, hfs_file * cat,
             uint32_t linkNum =
                 tsk_getu32(fs->endian, cat->std.perm.special.inum);
 
-            TSK_INUM_T target_cnid;     //  This is the real CNID of the file.
-
-            target_cnid = hfs_lookup_hard_link(hfs, linkNum, FALSE);
+            // It seems like the linkNum is always the same as the CNID
+            // returned by hfs_lookup_hard_link, which is very inefficient when a 
+            // datasource contains large numbers of linked files.
+            TSK_INUM_T target_cnid = linkNum;
 
             if (target_cnid != 0) {
```

Comment:
This check should likely be removed now.

Reply:
Is what you're saying is that `linkNum` should simply be returned here unchecked, as there's no possibility for having an error?